### DISCUSSION TITLE
Delete a whole folder from stroage

### DISCUSSION THREAD

**Discussion by madeleineostoja** [supabase/supabase#10048] [Questions] [answered]
Is there a way we can delete a folder from storage without having to loop through and deleting every item inside it first? My use case is account deletion, each of my users have a folder (mapped to their user id) with all their assets in it. I need to delete that before deleting the user otherwise I get foreign key errors on the `objects` schema.

I've tried `storage.from({bucket}).remove([{folder}])` but that doesn't seem to work

**Comment by GaryAustin1** [comment, marked as answer]
No.  There are several hits here and in storage-api issues on this.
https://github.com/supabase/storage-api/issues/207 (this implies something is in the works)
https://github.com/supabase/supabase/discussions/4218

**Comment by madeleineostoja** [comment]
Ah missed that issue, I only searched this repo. Thanks!
